provider/maas/environ.go

```diff
@@ -286,36 +253,13 @@ func (env *maasEnviron) SetCloudSpec(_ stdcontext.Context, spec environscloudspe
 		return errors.Trace(err)
 	}
 
-	// We need to know the version of the server we're on. We support 1.9
-	// and 2.0. MAAS 1.9 uses the 1.0 api version and 2.0 uses the 2.0 api
-	// version.
 	apiVersion := apiVersion2
 	controller, err := GetMAAS2Controller(maasServer, maasOAuth)
-	switch {
-	case gomaasapi.IsUnsupportedVersionError(err):
```

Comment:
We will still get this error type. 

_gomaasapi_ actually has only 2.0 as the supported version. So if we try to connect with `http://maas.server/MAAS` as the cloud URL, it will try to connect using that API. A future version of MAAS supporting only say 3.0 (note this is the API version; not the MAAS version) would return an error like: 
`controller at http://maas.server/MAAS does not support any of [2.0]`.

but in the event that you supplied something like `http://maas.server/MAAS/api/5.0/` as the cloud URL, it you would also get an `UnsupportedVersionError` regarding you implied version 5.0.

Reply:
So I think I understand that gomaasapi is doing the versioning checks for supported API and we don't have to worry about it at this level.

Reply:
That's correct.